phoenix-core/src/main/java/org/apache/phoenix/expression/function/InstrFunction.java

```diff
@@ -0,0 +1,128 @@
+package org.apache.phoenix.expression.function;
+
+import java.util.Arrays;
+import java.util.List;
+
+import org.apache.hadoop.hbase.io.ImmutableBytesWritable;
+import org.apache.phoenix.expression.Expression;
+import org.apache.phoenix.parse.FunctionParseNode.Argument;
+import org.apache.phoenix.parse.FunctionParseNode.BuiltInFunction;
+import org.apache.phoenix.schema.SortOrder;
+import org.apache.phoenix.schema.tuple.Tuple;
+import org.apache.phoenix.schema.types.PDataType;
+import org.apache.phoenix.schema.types.PInteger;
+import org.apache.phoenix.schema.types.PVarchar;
+import org.apache.phoenix.util.StringUtil;
+
+@BuiltInFunction(name=InstrFunction.NAME, args={
+        @Argument(allowedTypes={ PVarchar.class }),
+        @Argument(allowedTypes={ PVarchar.class })})
+public class InstrFunction extends ScalarFunction{
+    
+    public static final String NAME = "INSTR";
+    
+    private String strToSearch;
+    
+    public InstrFunction() { }
+    
+    public InstrFunction(List<Expression> children, String strToSearch) {
+        super(children);
+        setStrToSearch(strToSearch);
+    }
+    
+    
+    private Expression getStringExpression() {
+        return children.get(0);
+    }
+    
+    
+    @Override
+    public boolean evaluate(Tuple tuple, ImmutableBytesWritable ptr) {
+        Expression child = getStringExpression();
+        if (!child.evaluate(tuple, ptr)) {
+            return false;
+        }
+        
+        int position;
+        //Logic for Empty string search
+        if(strToSearch.length()==0){
+            position = 0;
+            ptr.set(PInteger.INSTANCE.toBytes(position));
+            return true;
+        }
+        
+        
+        byte [] strPattern = strToSearch.getBytes();
+        SortOrder sortOrder = child.getSortOrder();
+        ImmutableBytesWritable str = new ImmutableBytesWritable(strPattern);
+        
+        position = searchString(ptr,str,sortOrder,strPattern);
+        
+        ptr.set(PInteger.INSTANCE.toBytes(position));
+        return true;
+    }
+
+    @Override
+    public PDataType getDataType() {
+        return PInteger.INSTANCE;
+    }
+
+    @Override
+    public String getName() {
+        return NAME;
+    }
+
+    public String getStrToSearch() {
+        return strToSearch;
+    }
+
+    public void setStrToSearch(String strToSearch) {
+        this.strToSearch = strToSearch;
+    }
+    
+    //Below Method is to Search the Pattern in the given String
+    private int searchString(ImmutableBytesWritable ptr, ImmutableBytesWritable strbytes,
+                             SortOrder sortOrder, byte [] strPattern) {
+        
+        int textOffset = ptr.getOffset();
+        int textLength = ptr.getLength();
+        int strLength  = strbytes.getLength();
+        byte [] text = ptr.get();
+        byte[] target = new byte[strLength];
+        int position = 0;
+        
+        //Logic to compare Byte arrays
+        while((textLength - textOffset) > 0){
```

Comment:
Minor code formatting issue: there should be a space between the "while" and the opening parenthesis